### PR TITLE
Integrate new `WebSocket` host

### PR DESCRIPTION
**Pull request by miiu96**
- integrated the new WebSocket host from the repository `mx-chain-communication-go`
- update core

### CODE REVIEW

File: cmd/elasticindexer/config/prefs.toml [hunk 1]
```diff
@@ -2,12 +2,14 @@
     disabled-indices = []
     [config.web-socket]
         server-url = "localhost:22111"
+        is-server = true
```
Comment (bogdan-rosianu) [inline cmd/elasticindexer/config/prefs.toml:5]: maybe it should be `mode = "server"`

Comment (miiu96) [reply cmd/elasticindexer/config/prefs.toml]: fixed

File: cmd/elasticindexer/config/prefs.toml [hunk 2]
```diff
@@ -1,14 +1,18 @@
 [config]
     disabled-indices = []
     [config.web-socket]
-        server-url = "localhost:22111"
-        is-server = true
+        # URL for the WebSocket client/server connection
+        # This value represents the IP address and port number that the WebSocket client or server will use to establish a connection.
+        url = "localhost:22111"
+        # This flag will start the WebSocket connector as server or client( can be "client" or "server")
```
Comment (bogdan-rosianu) [inline cmd/elasticindexer/config/prefs.toml:7]: This flag describes the mode to start the WebSocket connector. Can be "client" or "server"

Comment (miiu96) [reply cmd/elasticindexer/config/prefs.toml]: done

**Review by bogdan-rosianu — DISMISSED** [review]

File: process/wsindexer/indexer.go [hunk 3]
```diff
@@ -42,25 +41,25 @@ func NewIndexer(marshaller marshal.Marshalizer, dataIndexer DataIndexer) (*index
...
+		outport.TopicSaveValidatorsRating:  i.saveValidatorsRating,
+		outport.TopicSaveValidatorsPubKeys: i.saveValidatorsPubKeys,
+		outport.TopicSaveAccounts:          i.saveAccounts,
+		outport.TopicFinalizedBlock:        i.finalizedBlock,
 	}
 }
 
-func (i *indexer) ProcessPayload(payload *data.PayloadData) error {
-	operationAction, ok := i.actions[payload.OperationType]
+func (i *indexer) ProcessPayload(payload []byte, topic string) error {
```
Comment (iulianpascalau) [inline process/wsindexer/indexer.go:55]: missing comment

Comment (miiu96) [reply process/wsindexer/indexer.go:55]: done.

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**